pennylane_qulacs/qulacs_device.py

```diff
@@ -39,181 +74,239 @@
         [0, 0, 0, np.exp(1j * theta / 2)],
     ]
 )
-toffoli = np.diag([1 for i in range(8)])
-toffoli[6:8, 6:8] = np.array([[0, 1], [1, 0]])
 
 
-def hermitian(*args):
-    r"""Input validation for an arbitary Hermitian expectation.
+def _reverse_state(state_vector):
+    """Reverse the qubit order for a vector of amplitudes.
     Args:
-        args (array): square hermitian matrix
+        state_vector (iterable[complex]): vector containing the amplitudes
     Returns:
-        array: square hermitian matrix
+        list[complex]
     """
-    A = np.asarray(args[0])
-
-    if A.shape[0] != A.shape[1]:
-        raise ValueError('Expectation must be a square matrix.')
+    state_vector = np.array(state_vector)
+    N = int(np.log2(len(state_vector)))
+    reversed_state = state_vector.reshape([2] * N).T.flatten()
+    return list(reversed_state)
```

Comment:
Does qulacs return the probability as a list? Strange, it would be a lot more memory efficient to use a NumPy array.

Reply:
No, I think this was more from us, I thought it is more compatible. I see now that default.qubit returns np arrays. So deleted the list conversion.